pkg/kn/commands/service/export.go

```diff
@@ -0,0 +1,233 @@
+// Copyright © 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"errors"
+
+	"sort"
+	"strconv"
+
+	"github.com/spf13/cobra"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"knative.dev/client/pkg/kn/commands"
+	clientservingv1 "knative.dev/client/pkg/serving/v1"
+	"knative.dev/serving/pkg/apis/serving"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+)
+
+// NewServiceExportCommand returns a new command for exporting a service.
+func NewServiceExportCommand(p *commands.KnParams) *cobra.Command {
+
+	// For machine readable output
+	machineReadablePrintFlags := genericclioptions.NewPrintFlags("")
+
+	command := &cobra.Command{
+		Use:   "export NAME",
+		Short: "export a service",
+		Example: `
+  # Export a service in yaml format
+  kn service export foo -n bar -o yaml
+  # Export a service in json format
+  kn service export foo -n bar -o json`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return errors.New("'kn service export' requires name of the service as single argument")
+			}
+			if !machineReadablePrintFlags.OutputFlagSpecified() {
+				return errors.New("'kn service export' requires output format")
+			}
+			serviceName := args[0]
+
+			namespace, err := p.GetNamespace(cmd)
+			if err != nil {
```

Comment:
Does it make sense to log error or some output to help end user?

Reply:
Returning the error should print it on the users screen.